client/src/pages/ChatPage.js

```diff
@@ -0,0 +1,315 @@
+                    <PaperAirplaneIcon className="h-3 w-3" />
+                  </IconButton>
+                </div>
+              </form>
+            </Card>
+          </div>
+        </div>
+      </div>
+
+      <Dialog
+        size="sm"
+        open={openUser}
+        handler={addUserHandler}
+        className="bg-transparent shadow-none"
+      >
+        <Card className="mx-auto w-full max-w-[24rem]">
+          <CardHeader
+            variant="gradient"
+            className="mb-4 grid h-28 place-items-center bg-indigo-300"
+          >
+            <Typography variant="h3" color="white">
+              Invite Friend
+            </Typography>
+          </CardHeader>
+          <CardBody className="flex flex-col gap-4">
+            <Input label="Username" size="lg" onChange={onInviteUser} />
+          </CardBody>
+          <CardFooter className="pt-0">
+            <Button
+              className="bg-indigo-300"
+              onClick={inviteDispatch}
+              fullWidth
+            >
+              Invite
+            </Button>
+          </CardFooter>
+        </Card>
+      </Dialog>
+
+      <Dialog
+        size="sm"
+        open={openName}
+        handler={changeNameHandler}
+        className="bg-transparent shadow-none"
+      >
+        <Card className="mx-auto w-full max-w-[24rem]">
+          <CardHeader
+            variant="gradient"
+            className="mb-4 grid h-28 place-items-center bg-indigo-300"
+          >
+            <Typography variant="h3" color="white">
+              Change Group Name
+            </Typography>
+          </CardHeader>
+          <CardBody className="flex flex-col gap-4">
+            <Input
+              label="Name"
+              size="lg"
+              value={name}
+              onChange={(e) => setName(e.target.value)}
+            />
+          </CardBody>
+          <CardFooter className="pt-0">
+            <Button className="bg-indigo-300" onClick={onChangeName} fullWidth>
+              Change
+            </Button>
+          </CardFooter>
+        </Card>
+      </Dialog>
+
+      <Dialog open={openExit} handler={exitUserHandler} size="sm">
+        <DialogHeader>Are you sure you want to leave this chat?</DialogHeader>
+        <DialogBody divider>
+          Please note that you'll only be able to join this chat agian through
+          an invitation.
+        </DialogBody>
+        <DialogFooter>
+          <Button
+            size="sm"
+            variant="text"
+            color="blue-gray"
+            className="flex items-center mr-2 border border-gray-400/70"
+            onClick={exitUserHandler}
+          >
+            Cancel
+          </Button>
+          <Button
+            size="sm"
+            variant="text"
+            color="blue-gray"
+            className="flex items-center border border-gray-400/70"
+            onClick={onExitUser}
+          >
+            Yes
```

Comment:
Perhaps in the future there should be some user feedback to other group members when a user joins or leaves the chat group.